hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAirdropValidator.java

```diff
@@ -152,23 +173,9 @@ private static void validateFungibleTransfers(
             final TokenID tokenId,
             final AccountAmount senderAmount,
             final ReadableTokenRelationStore tokenRelStore) {
-        final var tokenRel = tokenRelStore.get(senderAccount.accountIdOrThrow(), tokenId);
-        validateTrue(tokenRel != null, TOKEN_NOT_ASSOCIATED_TO_ACCOUNT);
-        if (senderAmount.isApproval()) {
-            final var tokenAllowances = senderAccount.tokenAllowances();
-            var haveExistingAllowance = false;
-            for (final var allowance : tokenAllowances) {
-                if (payer.equals(allowance.spenderId()) && tokenId.equals(allowance.tokenId())) {
-                    haveExistingAllowance = true;
-                    final var newAllowanceAmount = allowance.amount() + senderAmount.amount();
-                    validateTrue(newAllowanceAmount >= 0, AMOUNT_EXCEEDS_ALLOWANCE);
-                    break;
-                }
-            }
-            validateTrue(haveExistingAllowance, SPENDER_DOES_NOT_HAVE_ALLOWANCE);
-        } else {
-            validateTrue(tokenRel.balance() >= Math.abs(senderAmount.amount()), INVALID_ACCOUNT_AMOUNTS);
-        }
+        // validate association and account frozen
+        final var tokenRel = getIfUsable(senderAccount.accountIdOrThrow(), tokenId, tokenRelStore);
+        validateTrue(tokenRel.balance() >= Math.abs(senderAmount.amount()), INSUFFICIENT_TOKEN_BALANCE);
```

Comment:
Why `Math.abs` ?

Reply:
Because for the sender the value in accountAmounts is negative.